raiden/transfer/identifiers.py

```diff
@@ -39,3 +32,38 @@ def validate(self) -> None:
 
         if self.channel_identifier < 0 or self.channel_identifier > constants.UINT256_MAX:
             raise ValueError("channel id is invalid")
+
+    def __str__(self) -> str:
+        return (
+            f"{self.chain_identifier}|{to_checksum_address(self.token_network_address)}|"
+            f"{self.channel_identifier}"
+        )
+
+
+def string_to_canonical_identifier(string: str) -> CanonicalIdentifier:
```

Comment:
My previous request was to move this function to the fields.py module... `__str__` also should be there as well since this is only serialization-related.